### ISSUE TITLE
Improve FHIR version error message

### ISSUE THREAD

**Issue by wardweistra** [FirelyTeam/firely-net-sdk#1301]
Both in Simplifier and Forge the error message of the API regarding FHIR version is directly shown, but unintuitive to users. Currently it will state: "While building a POCO: Literal '4.0.1' is not a valid value for enumeration 'FHIRVersion' (at ...)". However, 4.0.1 is a valid FHIR version, just not supported in this API yet.

Please change to something that reflects that, like: _The FHIR version 4.0.1 is not (yet) supported in this version (1.x) of the .NET FHIR API. Please see the supported versions of your product or of the latest version of the API at ..._

**Simplifier:**
https://simplifier.net/fishtesting/fishpatient
First load says can't render (separate issue), second time shows: 
The resource information may be incomplete due to the following issues:
* Invalid resource format: While building a POCO: Literal '4.0.1' is not a valid value for enumeration 'FHIRVersion' (at StructureDefinition.fhirVersion[0])

**Forge:**
We've tried to publish the profiles we've built using Forge: The Forge menu item "Publish to FHIR Server" lead us to a dialogue window where we can select a FHIR Server. However, the "Connect"-Button returned the error message "While building a POCO: Literal '4.0.1' is not a valid value for enumeration 'FHIRVersion' (at CapabilityStatement.fhirVersion[0])"." This error message is wrong / misleading, as '4.0.1' is listed in http://hl7.org/fhir/valueset-FHIR-version.html.

**Comment by ewoutkramer** [comment]
It's not intuitive since at that point in the software it has no idea its dealing with versions or talking about versions.  You'll get the same message when you try to put the value 'femalee' in the gender field:  Literal 'femalee' is not a valid value for enumeration 'AdminstrativeGender'.

But if this is confusing we need to fix this at a different level. E.g. before even trying to call the parser for StructureDefinitions (in this case) we should check whether the library and the version of the data matches.  This might actually be simplest in Simplifier itself, since it knows the data AND the version of the API it is using, even before it invokes our code.

**Comment by vivektk2020** [comment]
Getting the same error while trying to load a valid  R4 profile. I am using an old version of Hl7.Fhir.Specification.R4  package(version 1.4).Does it mean that the library does not support the loading of R4 profiles?

**Comment by marcovisserFurore** [comment]
Hi @vivektk2020 , 
The `FhirVersion` of a `StructureDefinition` (Profile) is of type `code` and is bound by a set of [predefined values](https://www.hl7.org/fhir/valueset-FHIR-version.html). Just like the resource `Patient` with `AdminstrativeGender`. See the [example](https://github.com/FirelyTeam/firely-net-sdk/issues/1301#issuecomment-600001571) of @ewoutkramer above.

You are using version 1.4 of this SDK. When we released this version (Oct 10, 2019) version 4.0.1 was not published yet, so our SDK does not know about this. Since version [SDK 1.5](https://github.com/FirelyTeam/firely-net-sdk/releases/tag/v1.5.0-r4) we updated the newest FHIR version for R4 (4.0.1).

So updating the `Hl7.Fhir.Specification.R4` package to version 1.5 or higher will solve your problem.
